File: frontend/src/types/quotation.ts
export type TipoOperacion =
  | "EM"
  | "IM"
  | "EA"
  | "IA"
  | "ET"
  | "IT"
  | "CO";

export type QuotationForm = {
  referencia: string;
  tipo_operacion: TipoOperacion | "";

  // Sección 2: Ruta y Transporte
  pol?: string;
  transbordo_1?: string;
  transbordo_2?: string;
  transbordo_3?: string;
  pod?: string;
  linea_maritima?: string;
  aerolinea?: string;
  transit_time?: number;
  dias_libres_destino?: number;
  incoterm_origen: string;
  incoterm_destino: string;
  lugar_pickup: string;
  lugar_delivery: string;
  dias_libres_origen: number;

  // Sección 3: Detalles de Mercadería
  mercaderia?: string;
  bultos?: number;
  peso_bruto_kg?: number;
  toneladas?: number;
  volumen_cbm?: number;
  chargeable_weight?: number;
  largo?: number;
  ancho?: number;
  alto?: number;
  valor_comercial?: number;

   // Sección 4: Equipo y Especificaciones
  tipo_carga?: "FCL" | "LCL";
  tipo_contenedor?: string;
  cantidad_contenedores?: number;
  cantidad_bls?: number;

  apto_alimento?: boolean;
  tiene_hielo_seco?: boolean;
  validez_dias?: number;
};

File: backend/app/schemas/quotation.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime

class QuotationCreate(BaseModel):
    referencia: str
    tipo_operacion: str
    pol: Optional[str] = None
    transbordo_1: Optional[str] = None
    transbordo_2: Optional[str] = None
    transbordo_3: Optional[str] = None
    pod: Optional[str] = None
    linea_maritima: Optional[str] = None
    aerolinea: Optional[str] = None
    transit_time: Optional[float] = None
    dias_libres_destino: Optional[float] = None
    mercaderia: Optional[str] = None
    bultos: Optional[int] = 1
    peso_bruto_kg: Optional[float] = None
    toneladas: Optional[float] = None
    volumen_cbm: Optional[float] = None
    chargeable_weight: Optional[float] = None
    largo: Optional[float] = None
    ancho: Optional[float] = None
    alto: Optional[float] = None
    valor_comercial: Optional[float] = None
    tipo_carga: Optional[str] = None
    tipo_contenedor: Optional[str] = None
    cantidad_contenedores: Optional[int] = 1
    cantidad_bls: Optional[int] = 1
    apto_alimento: Optional[bool] = False
    tiene_hielo_seco: Optional[bool] = False
    validez_dias: Optional[int] = None
    
    # Añade estos campos que están en tu modelo pero faltan en el schema:
    incoterm_origen: Optional[str] = None
    incoterm_destino: Optional[str] = None
    lugar_pickup: Optional[str] = "N/A"
    lugar_delivery: Optional[str] = "N/A"

     # Nuevos campos
    incoterm_origen: Optional[str] = None
    incoterm_destino: Optional[str] = None
    lugar_pickup: Optional[str] = "N/A"
    lugar_delivery: Optional[str] = "N/A"
    dias_libres_origen: Optional[int] = None
    
    class Config:
        # Esto permite que campos extra no causen error
        extra = "ignore"


class QuotationResponse(BaseModel):
    id: str
    codigo_tracking: str
    referencia: str
    tipo_operacion: str
    pol: Optional[str]
    pod: Optional[str]
    incoterm_origen: Optional[str]
    incoterm_destino: Optional[str]
    mercaderia: Optional[str]
    bultos: Optional[int]
    peso_bruto_kg: Optional[float]
    volumen_cbm: Optional[float]
    tipo_carga: Optional[str]
    tipo_contenedor: Optional[str]
    created_at: datetime
    updated_at: Optional[datetime]
    
    class Config:
        orm_mode = True
        from_attributes = True